### PR TITLE
[Refactor] Simplify and decouple the syntax of GrantPriv in RBAC

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Signed-off-by: HangyuanLiu <460660596@qq.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4 [hunk 1]
```diff
@@ -254,6 +254,7 @@ PLUGIN: 'PLUGIN';
 PLUGINS: 'PLUGINS';
 PRECEDING: 'PRECEDING';
 PRIMARY: 'PRIMARY';
+PRIVILEGES: 'PRIVILEGES';
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:257]: have we used `PRIVILEGES` as a keyword in `StarRocks.g4`?

**Comment by nshangyiming** [comment]
run starrocks_admit_test

**Comment by nshangyiming** [comment]
conflict

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 266 / 281 (94.66%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|2|00.00%|[675, 681]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToStringBuilder.java|3|4|75.00%|[378]|
|:large_blue_circle:|com/starrocks/privilege/ResourceGroupPEntryObject.java|5|6|83.33%|[46]|
|:large_blue_circle:|com/starrocks/privilege/MaterializedViewPEntryObject.java|12|14|85.71%|[52, 53]|
|:large_blue_circle:|com/starrocks/sql/ast/BaseGrantRevokePrivilegeStmt.java|7|8|87.50%|[101]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzerV2.java|30|34|88.24%|[243, 257, 299, 307]|
|:large_blue_circle:|com/starrocks/privilege/FunctionPEntryObject.java|13|14|92.86%|[64]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|115|118|97.46%|[3167, 4083, 4084]|
|:large_blue_circle:|com/starrocks/privilege/UserPEntryObject.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/privilege/AuthUpgrader.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/GrantRevokePrivilegeObjects.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/GrantPrivilegeStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/ResourcePEntryObject.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/DbPEntryObject.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/privilege/Auth.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RevokePrivilegeStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/ViewPEntryObject.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/GlobalFunctionPEntryObject.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/TablePEntryObject.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/CatalogPEntryObject.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4 (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:257]: ALL is an abbreviation for ALL PRIVILEGES

**Review by nshangyiming — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Pull request merged**